Sources/MapboxMaps/Foundation/Tracing.swift

```diff
@@ -0,0 +1,96 @@
+import Foundation
+import os.signpost
+import MapboxCommon
+
+private let subsystem = "com.mapbox.maps"
+
+internal extension OSLog {
+    @available(iOS 12, *)
+    private static let _poi = OSLog(subsystem: subsystem, category: .pointsOfInterest)
+    private static let _platform = OSLog(subsystem: subsystem, category: "platform")
+    private static let tracingEnabled: Bool = SettingsServiceFactory.getInstance(storageType: .nonPersistent).isTracingEnabled
+
+    static let platform: OSLog = tracingEnabled ? _platform : .disabled
+    static let poi: OSLog = {
+        if #available(iOS 12, *), tracingEnabled {
+            return ._poi
+        } else {
+            return .disabled
+        }
+    }()
+
+    func beginInterval(name: StaticString, beginMessage: String? = nil) -> SignpostInterval {
+        let interval = SignpostInterval(log: self, name: name)
+        interval.begin(message: beginMessage)
+        return interval
```

Comment:
Discussed with @persidskiy non-zero impact of creating a struct, closure and capturing variables by the closure in production. In the spectrum of zero, almost-zero, infinity - we deemed it to have almost-zero impact on production code.